src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TypeChar.cs

```diff
@@ -357,12 +372,12 @@ private bool IsFilterCharacter(char ch)
             return GetCompletionRules().IsFilterCharacter(selectedItem, ch, filterText);
         }
 
-        private string GetCurrentFilterText(Model model, CompletionItem selectedItem)
+        private string GetCurrentFilterText(Model model, CompletionItem selectedItem, bool stopBeforeCaret = true)
         {
             var textSnapshot = this.TextView.TextSnapshot;
             var viewSpan = model.GetSubjectBufferFilterSpanInViewBuffer(selectedItem.FilterSpan);
             var filterText = model.GetCurrentTextInSnapshot(
-                viewSpan, textSnapshot, GetCaretPointInViewBuffer());
+                viewSpan, textSnapshot, endPoint: stopBeforeCaret ? (int?)GetCaretPointInViewBuffer() : null);
```

Comment:
After playing with both VS2015 and VS2013, I find that:

- When Ctrl+J was fired at the beginning of word, they both have empty filter span, so wrapping functions works well in both versions of VS.
- When Ctrl+J was fired in the middle of word, they both have the word in filter span, VS2013 replace the word when commit character typed, if the caret is NOT at the beginning; while VS2015 always replace the span between the start of the word and caret with the selected item when you use commit character, BUT always replace the whole word when you hit Enter.

I think the VS2015 behavior is arguably better. Consider when you first type something at the beginning of a word but no completion list shows, e.g. you have the following methods:
```C#
int Foo()
{
    return |Foo();
}

int BarVeryLong(int a) { return 0; }
```
Then you type `ba`, no completion list here, instead VS guesses you want to generate a new function later. So you have to press Ctrl+J, but now, the filter span will be the whole `baFoo`. When you find the `BarVeryLong` in the list and type `(`, VS2015 give you `BarVeryLong(Foo();`, VS2013 give you `BarVeryLong(();`. For this case, VS2015 wins!